src/traces/scattergl/attributes.js

```diff
@@ -54,6 +54,8 @@ var attrs = module.exports = overrideAll({
     line: {
         color: scatterLineAttrs.color,
         width: scatterLineAttrs.width,
+        shape: scatterLineAttrs.shape,
+        smoothing: scatterLineAttrs.smoothing,
```

Comment:
Ha. I see what you did there. You had to add `smoothing` so that you could reuse `handleLineShapeDefaults` in the defaults directly.

But `smoothing` and `line.shape: 'spline'` aren't implemented in scattergl, so we can't just reuse the scatter attributes and defaults. It's important to have the correct attributes declaration as these are used to generate doc pages like this one: https://plot.ly/javascript/reference/

Reply:
OK ... so i will copy/paste instead .